minetest-mods/technic
# technic_cnc: use programs definition to generate formspec

This deprecates the variables onesize_products and twosize_products by using a shared definition table.
With these changes, it should be easier to add new programs in the future.

New feature: Tooltips are added to all programs. (only applies to newly placed nodes)

**Before:**

![grafik](https://github.com/user-attachments/assets/9b543eea-29de-4120-b6f2-9657f159e9e6)

**After:**

![grafik](https://github.com/user-attachments/assets/19b5724a-e234-4dc3-8f08-26d168abcc2c)

**How to test:**

1. Place a CNC using the current technic master version
2. Ensure that it the CNC cuts properly
3. Apply this PR
4. Ensure that it the CNC cuts properly
5. Place a new CNC
6. Ensure that it the CNC cuts properly
